### PR TITLE
scheduler: add numOfNodesToScore to the scheduler profile

### LINKED ISSUE #86630: Scheduler: use the power of two random choices to select nodes
**What would you like to be added**:

[The power of two random choices](https://www.eecs.harvard.edu/~michaelm/postscripts/handbook2001.pdf) has been used in many systems, e.g. [Sparrow: Distributed, Low Latency Scheduling](https://cs.stanford.edu/~matei/papers/2013/sosp_sparrow.pdf), [Nomad](https://www.nomadproject.io/docs/schedulers.html#batch). Kube-schedule could also use it to to limit the number of nodes that are ranked in order to reduce the time for running score plugins significantly.

The main idea is that scheduler chooses two nodes randomly from those nodes filered by filter plugins, then run socre plugins only for the two nodes.

We can implement it as a post-filter plugins with some modification(e.g. `postfilterStatus := g.framework.RunPostFilterPlugins(ctx, state, pod, filteredNodes, filteredNodesStatuses)` => `posfilteredNodes, postfilterStatus := g.framework.RunPostFilterPlugins(ctx, state, pod, filteredNodes, filteredNodesStatuses)`)

**Why is this needed**:

It is very usefully for the batch workload since they are short lived and not so sensitive to that whether it is really the best choice node, by using it kube-scheduler could be used as a batch workload scheduler with higher throughput. And it is also useful for the service workload if they are not so senstive to best choice node.

/sig scheduling
/assign

### PR DESCRIPTION
**Pull request by tzzcfrank**
#### What type of PR is this?
/kind feature

#### What this PR does / why we need it:
scheduler: add numOfNodesToScore to the scheduler profile

#### Which issue(s) this PR fixes:

Fixes #86630
Ref   #95709  #93270 

#### Does this PR introduce a user-facing change?

```release-note
scheduler: support profile-level NumOfNodesToScore parameter
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

File: pkg/scheduler/apis/config/validation/validation_test.go [hunk 1]
```diff
@@ -475,6 +476,46 @@ func TestValidateKubeSchedulerConfigurationV1beta3(t *testing.T) {
...
+	negativeIntNumOfNodesToScoreVal := intstr.FromInt(-10)
+	negativeIntNumOfNodesToScore.Profiles[0].NumOfNodesToScore = &negativeIntNumOfNodesToScoreVal
+
+	zeroIntNumOfNodesToScore := validConfig.DeepCopy()
+	zeroIntNumOfNodesToScoreVal := intstr.FromInt(0)
+	zeroIntNumOfNodesToScore.Profiles[0].NumOfNodesToScore = &zeroIntNumOfNodesToScoreVal
+
+	normalIntNumOfNodesToScore := validConfig.DeepCopy()
+	normalIntNumOfNodesToScoreVal := intstr.FromInt(50)
+	normalIntNumOfNodesToScore.Profiles[0].NumOfNodesToScore = &normalIntNumOfNodesToScoreVal
```
Comment (BinacsLee) [inline pkg/scheduler/apis/config/validation/validation_test.go:517]: nit: shall we abstract out a function to make the code more concise?

Comment (tzzcfrank) [reply pkg/scheduler/apis/config/validation/validation_test.go]: Done

**Comment by Huang-Wei** [comment]
@tzzcfrank thanks for the PR, however, we haven't come to a consensus yet at the original issue #86630.

**Pull request closed without merging**